C-Plus-Plus/dp/Zero_N_Knapsack.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+You've heard of 0-1 knapsack. Below is the problem statement for the same.
+
+Given weights and values of n items, put these items in a knapsack of capacity cap to get the maximum total value in the knapsack. In other words, given two integer arrays val[0..n-1] and wt[0..n-1] which represent values and weights associated with n items respectively. Also given an integer cap which represents knapsack capacity, find out the maximum value subset of val[] such that sum of the weights of this subset is smaller than or equal to cap. You cannot break an item, either pick the complete item, or don�t pick it (0-1 property).
```

Comment:
Please format the description in such a way that the whole description is readable on a phone or a laptop screen at once. And please also add a description of your approach to solve this problem.